submissions/Mifaresss/tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,63 @@
+class Inhabitants {
+   constructor(name, species, gender, saying) {
+      this.name = name;
+      this.species = species;
+      this.gender = gender;
+      this.saying = saying;
+   }
+   prepareforPrinting() {
+      return [this.name, this.species, this.gender, this.saying].join('; ');
+   }
+}
+
+class Humans extends Inhabitants {
+   constructor(name, gender, saying) {
+      super(name, 'human', gender, saying);
+      this.legs = 2;
+      this.hands = 2;
+   }
+   prepareforPrinting() {
+      return [super.prepareforPrinting(), this.legs, this.hands].join('; ');
+   }
+}
+
+class Animals extends Inhabitants {
+   constructor(name, species, gender, saying) {
+      super(name, species, gender, saying);
+      this.legs = 4;

```

Comment:
Why `name` is generalized into the base class and `legs` are not?
Not a rhetoric question.

Reply:
Because everyone has a name, but no legs (fish, for example)

Reply:
Initially, I did not add legs to the base class, since not everyone has them (fish, for example). But now it doesn't seem logical. Understood the error. I'll fix it now

Reply:
done)